packages/x-tree-view/src/internals/plugins/useTreeViewFocus/useTreeViewFocus.ts

```diff
@@ -107,11 +107,13 @@ export const useTreeViewFocus: TreeViewPlugin<UseTreeViewFocusSignature> = ({
     }
 
     const node = instance.getNode(state.focusedNodeId);
-    const itemElement = document.getElementById(
-      instance.getTreeItemId(state.focusedNodeId, node.idAttribute),
-    );
-    if (itemElement) {
-      itemElement.blur();
+    if (node) {
```

Comment:
Not directly related but I noticed this bug while writing the tests.
If we remove the currently focused item from `props.items`, then its `TreeItem`  while call `removeFocusedItem`, but the state does not contain the item anymore so it crashes.